include/nicegraf.h

```diff
@@ -3306,6 +3306,114 @@ void ngf_cmd_write_image(
  */
 ngf_error ngf_cmd_generate_mipmaps(ngf_xfer_encoder xfenc, ngf_image img) NGF_NOEXCEPT;
 
+/**
+ * \ingroup ngf
+ * The RenderDoc API version to be used
+ */
+#define NGF_RENDERDOC_API_VERSION 1_6_0
+
+/**
+ * \ingroup ngf
+ * Helper to obtain a RenderDoc API interface with \ref NGF_RENDERDOC_API_VERSION. 
+ */
+#define NGF_RENDERDOC_API NGFI_EVAL_AND_PASTE(RENDERDOC_API_, NGF_RENDERDOC_API_VERSION)
+
+/**
+ * \ingroup ngf
+ * Helper to obtain a RenderDoc API enum with \ref NGF_RENDERDOC_API_VERSION. 
+ */
+#define NGF_RENDERDOC_VERSION NGFI_EVAL_AND_PASTE(eRENDERDOC_API_Version_, NGF_RENDERDOC_API_VERSION)
+
+/**
+ * @struct ngf_rdoc_api
+ * \ingroup ngf
+ *
+ * A wrapper over a RenderDoc API instance.
+ *
+ * RenderDoc can be used to debug graphical programs. This interface
+ * allows you to perform frame captures programatically, rather than
+ * from the RenderDoc GUI.
+ *
+ * Through the RenderDoc API you can:
+ *  - Begin capturing a frame
+ *  - End capturing a frame
+ *  - Change the location of where captures are written to
+ *  - Change the title of an in-progress capture
+ *  - Capture the next frame
+ *
+ * See also: \ref ngf_rdoc_init, \ref ngf_rdoc_capture_begin, \ref ngf_rdoc_capture_end
+ * */
+typedef struct ngf_rdoc_api_t* ngf_rdoc_api; 
+
+/**
+ * \ingroup ngf
+ * Initializes RenderDoc API.
+ *
+ * Dynamically loads the RenderDoc RENDERDOC_GetAPI function and returns a 
+ * ngf_doc_api_t wrapper. 
+ *
+ * @param result ngf_doc_api_t handle to write to.
+ * @param lib_path the relative (to process) or absolute path to the RenderDoc library.
+ */
+ngf_error ngf_rdoc_init(ngf_rdoc_api result, char* lib_path);
```

Comment:
I would prefer if we initialized the renderdoc api at context creation time. Keep the pointer to renderdoc API dispatch table in the context state. Add a new field `renderdoc_lib_path` to `ngf_init_info` and only initialize capture if that field is not NULL.